### PR TITLE
test: Test stored time

### PR DESCRIPTION
**Pull request by seblapela**
Test that integration_events and time_series_points does not include data with stored time after snapshot time

* #32

### CODE REVIEW

File: source/databricks/package/balance_fixing_total_production.py [hunk 1]
```diff
@@ -86,6 +85,22 @@ def calculate_balance_fixing_total_production(
...
+        raw_integration_events_df.where(col("storedTime") <= batch_snapshot_datetime)
+        .withColumn("body", col("body").cast("string"))
+        .cache()
+    )
+
+
+def _get_time_series_points(
+    raw_time_series_points_df, batch_snapshot_datetime
+) -> DataFrame:
+    return raw_time_series_points_df.where(col("storedTime") <= batch_snapshot_datetime)
```
Comment (BjarkeMeier) [inline source/databricks/package/balance_fixing_total_production.py:101]: It doesn't origin at this PR, but shouldn't we ensure that we only use the latest time series point by registration date time?

File: source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            .select("storedTime", "body")
+        )
+
+    return factory
+
+
+def test__raw_events_with_stored_time_of_after_snapshot_time_is_not_included_in_cacheched_integration_events(
+    integration_events_df_factory, source_path
+):
+    events_df_first_of_june = integration_events_df_factory(stored_time=first_of_june)
```
Comment (BjarkeMeier) [inline source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py:96]: Can we add AAA comments?

File: source/databricks/package/balance_fixing_total_production.py (hunk 1 above)
Comment (seblapela) [reply source/databricks/package/balance_fixing_total_production.py:101]: This is done in _get_enriched_timeseries

File: source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py (hunk 2 above)
Comment (seblapela) [reply source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py:96]: done

**Review by lasrinnil — APPROVED** [review]

File: source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    events_df_second_of_june = integration_events_df_factory(stored_time=second_of_june)
+    events_df_third_of_june = integration_events_df_factory(stored_time=third_of_june)
+
+    # events from first second and third of June
+    events_df = events_df_first_of_june.union(events_df_second_of_june).union(
+        events_df_third_of_june
+    )
+
+    # assert that only events from first of june and before is returned when snapshot time is first of june
+    assert _get_cached_integration_events(events_df, first_of_june).count() == 1
```
Comment (BjarkeMeier) [inline source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py:106]: Can we make the tests parameterized and with only a single assert in each?

File: source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                        col("SettlementMethod"),
+                        col("ConnectionState"),
+                        col("EffectiveDate"),
+                        col("MeteringPointType"),
+                        col("MeteringPointId"),
+                        col("Resolution"),
+                        col("OperationTime"),
+                        col("CorrelationId"),
+                    )
+                ),
```
Comment (BjarkeMeier) [inline source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py:85]: This is not used in the test. Can we make the tests more explicit and simplify maintenance by removing this (or add an empty dummy body column if required)?

File: source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+# Factory defaults
+grid_area_code = "805"
+grid_area_link_id = "the-grid-area-link-id"
+gsrn_number = "the-gsrn-number"
+metering_point_id = "the-metering-point-id"
+
+first_of_june = datetime.strptime("31/05/2022 22:00", "%d/%m/%Y %H:%M")
+second_of_june = first_of_june + timedelta(days=1)
+third_of_june = first_of_june + timedelta(days=2)
```
Comment (BjarkeMeier) [inline source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py:43]: I guess we can remove most of these.

Comment (ftoft) [reply source/databricks/tests/integration/calculator/balance_fixing_total_production/test_get_cached_integration_events.py]: is done

**Pull request merged**